gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -392,112 +303,16 @@ public long getExpectedRecordCount() {
 
   @Override
   public void close() throws IOException {
-    if (currentPartitionIdx != INITIAL_PARTITION_IDX) {
-      updateStatisticsForCurrentPartition();
+    if (!allPartitionsFinished()) {
```

Comment:
Yes, the current implementation is confusing when end of partitions is reached. It calls updateStatisticsForCurrentPartition(), but essentially does nothing inside the method, since recordCount == 0. The change IMO is more readable.